Parser/Tree_Nodes/r_asmt.c

```diff
@@ -0,0 +1,96 @@
+/**************************************************************************************************
+*
+* File Name: r_asmt.c
+*
+* Author: Jameson Toper
+*
+* Created for CS344 on: 11/08/2019
+*
+* Purpose: code that provides functions for manipulating r_asmt nodes.
+*
+**************************************************************************************************/
+
+
+#include <stdio.h>
+#include "r_asmt.h"
+
+r_asmt *create_r_asmt(GArray *tokenStream, unsigned long index, unsigned long *next) {
+    /**
+     * Token[index] is ID
+     * Token[index+1] is '='
+     * Token[index+2 <--> index+n] is expression, need to confirm result is same type
+     * Token[index+n+1] is ';'
+     */
+
+    r_asmt *new_asmt = calloc(1, sizeof(r_asmt));
+
+    // Need to try expression before we add the ID, because it cannot be self-referencing
+    new_asmt->expression = create_expr(tokenStream, index + 2, next);
+
+    Type *varType = malloc(sizeof(Type));
```

Comment:
Unneeded and causes a memory leak, you can just create a `Type` variable and reference it in the function call, that way it will be removed from the stack frame when the function returns